src/metaopt/io/database/__init__.py

```diff
@@ -106,14 +106,14 @@ def write(self, collection_name, data,
         pass
 
     @abstractmethod
-    def read(self, collection_name, query, selection=None):
+    def read(self, collection_name, query=None, selection=None):
```

Comment:
So a call on read using an empty `query` means that we retrieve all of the documents inside? [but yeah we did agree on that]

Reply:
Oh damn I am not sure why I did that... see bad commit names. 😛 I am removing/stashing the commit tsirif@4570cd6 until it becomes necessary. @bouthilx agree?

Reply:
I don't understand what is the problem here. The behavior ```db.read("collection_name")``` being similar to ```db.read("collection_name", {})``` seems fine the me.